plugins/modules/aci_l3out_logical_interface_profile_eigrp_profile.py

```diff
@@ -0,0 +1,319 @@
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        l3out=dict(type="str", aliases=["l3out_name"]),
+        node_profile=dict(type="str", aliases=["node_profile_name", "logical_node"]),
+        interface_profile=dict(type="str", aliases=["interface_profile_name", "logical_interface"]),
+        eigrp_profile=dict(type="str", aliases=["name", "eigrp_profile_name"]),
+        eigrp_policy=dict(type="str", aliases=["eigrp_policy_name"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "l3out", "node_profile", "interface_profile"]],
+            ["state", "present", ["tenant", "l3out", "node_profile", "interface_profile", "eigrp_profile"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    l3out = module.params.get("l3out")
+    node_profile = module.params.get("node_profile")
+    interface_profile = module.params.get("interface_profile")
+    eigrp_profile = module.params.get("eigrp_profile")
+    eigrp_policy = module.params.get("eigrp_policy")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="l3extOut",
+            aci_rn="out-{0}".format(l3out),
+            module_object=l3out,
+            target_filter={"name": l3out},
+        ),
+        subclass_2=dict(
+            aci_class="l3extLNodeP",
+            aci_rn="lnodep-{0}".format(node_profile),
+            module_object=node_profile,
+            target_filter={"name": node_profile},
+        ),
+        subclass_3=dict(
+            aci_class="l3extLIfP",
+            aci_rn="lifp-[{0}]".format(interface_profile),
+            module_object=interface_profile,
+            target_filter={"name": interface_profile},
+        ),
+        subclass_4=dict(
+            aci_class="eigrpIfP",
+            aci_rn="eigrpIfP",
+            module_object=interface_profile,
+            target_filter={"name": interface_profile},
+        ),
```

Comment:
if eigrp interface profile is needed then should we not filter on that here?

```suggestion
        subclass_4=dict(
            aci_class="eigrpIfP",
            aci_rn="eigrpIfP",
            module_object= eigrp_profile,
            target_filter={"name": eigrp_profile},
        ),
```

Reply:
I removed eigrp_profile as an argument per the suggestion above, as such the interface_profile filter should now be correct.